packages/babel-plugin/src/utils/ast.tsx

```diff
@@ -99,6 +130,98 @@ export const getKey = (node: t.Expression) => {
   throw new Error();
 };
 
+/**
+ * Returns `true` if an identifier or any paths that reference the identifier are mutated.
+ * @param path
+ */
+const isIdentifierReferencesMutated = (path: NodePath<t.Identifier>): boolean => {
+  const binding = path.scope.getBinding(path.node.name);
+  if (!binding) {
+    return false;
+  }
+
+  if (!t.isVariableDeclarator(binding.path.node) || !binding.constant) {
+    return true;
+  }
+
+  for (let i = 0; i < binding.referencePaths.length; i++) {
+    const refPath = binding.referencePaths[i];
+    const innerBinding = refPath.scope.getOwnBinding(path.node.name);
+    if (!innerBinding) {
+      continue;
+    }
+
+    if (!t.isVariableDeclarator(innerBinding.path.node) || !innerBinding.constant) {
+      return true;
+    }
+  }
```

Comment:
nit: Can be replaced with a function to remove duplicate functionality if possible. Not sure. I prefer duplicate code over wrong abstraction any day. See if its really duplicate, otherwise lets skip abstraction part.

If we want, may be something like this

```typescript
function someFunction(path, getBindingFn) {
    const binding = path.scope[getBindingFn(path.node.name)];

    if (!binding) {
      return false;
    }

    if (!t.isVariableDeclarator(binding.path.node) || !binding.constant) {
      return true;
    }

    return false;
}
```

Reply:
i dont think its worth it for now to be honest, the two are slightly different

would recommend just leaving it for now

Reply:
yeeee duplicate code over wrong abstraction wins 😛 hurrayy 🎉